src/components/deck-gl-wrapper/custom-tile-3d-layer.ts

```diff
@@ -0,0 +1,575 @@
+    if (info.sourceLayer) {
+      info.sourceLayer.updateAutoHighlight(info);
+    }
+  }
+
+  private async _loadTileset(tilesetUrl) {
+    const { loadOptions = {} } = this.props;
+
+    // TODO: deprecate `loader` in v9.0
+    let loader = this.props.loader || this.props.loaders;
+    if (Array.isArray(loader)) {
+      loader = loader[0];
+    }
+
+    const options = { loadOptions: { ...loadOptions } };
+    if (loader.preload) {
+      const preloadOptions = await loader.preload(tilesetUrl, loadOptions);
+
+      if (preloadOptions.headers) {
+        options.loadOptions.fetch = {
+          ...options.loadOptions.fetch,
+          headers: preloadOptions.headers,
+        };
+      }
+      Object.assign(options, preloadOptions);
+    }
+    const tilesetJson = await load(tilesetUrl, loader, options.loadOptions);
+
+    const tileset3d = new Tileset3D(tilesetJson, {
+      onTileLoad: this._onTileLoad.bind(this),
+      onTileUnload: this._onTileUnload.bind(this),
+      onTileError: this.props.onTileError,
+      // New code ------------------
+      onTraversalComplete: this._onTraversalComplete.bind(this),
+      // ---------------------------
+      ...options,
+    });
+
+    this.setState({
+      tileset3d,
+      layerMap: {},
+    });
+
+    this._updateTileset(this.state.activeViewports);
+    this.props.onTilesetLoad(tileset3d);
+  }
+
+  private _onTileLoad(tileHeader: Tile3D): void {
+    const { lastUpdatedViewports } = this.state;
+    // New code ------------------
+    if (this.state.isCustomColors) {
+      this._colorizeTiles([tileHeader]);
+    }
+    // ---------------------------
+    this.props.onTileLoad(tileHeader);
+    // New code ------------------ condition is added
+    if (!this.state.colorsByAttribute) {
+      // ---------------------------
+      this._updateTileset(lastUpdatedViewports);
+      this.setNeedsUpdate();
+      // New code ------------------
+    }
+    // ------------------
+  }
+
+  private _onTileUnload(tileHeader: Tile3D): void {
+    // Was cleaned up from tileset cache. We no longer need to track it.
+    delete this.state.layerMap[tileHeader.id];
+    this.props.onTileUnload(tileHeader);
+  }
+
+  private _updateTileset(
+    viewports: { [viewportId: string]: Viewport } | null
+  ): void {
+    if (!viewports) {
+      return;
+    }
+    const { tileset3d } = this.state;
+    const { timeline } = this.context;
+    const viewportsNumber = Object.keys(viewports).length;
+    if (!timeline || !viewportsNumber || !tileset3d) {
+      return;
+    }
+    tileset3d.selectTiles(Object.values(viewports)).then((frameNumber) => {
+      const tilesetChanged = this.state.frameNumber !== frameNumber;
+      if (tilesetChanged) {
+        this.setState({ frameNumber });
+      }
+    });
+  }
+
+  // New code -------------------------------------------
+  private _onTraversalComplete(selectedTiles: Tile3D[]): Tile3D[] {
+    if (
+      this.state.isCustomColors &&
```

Comment:
I think about how to get rid of `isCustomColor` state... It looks unnecessary. Can we perform `this._colorizeTiles` in case of `customizeColors` function is defined? If not, don't do `_customizeColors`. It would be better then support some synthetic state. And you could check it and `TILE_TYPE.MESH` in one place, in the `this._colorizeTile` method.